webapp/channels/src/components/suggestion/generic_channel_provider.tsx

```diff
@@ -7,86 +7,67 @@ import {Channel} from '@mattermost/types/channels';
 import {ServerError} from '@mattermost/types/errors';
 import {ActionResult} from 'mattermost-redux/types/actions';
 
-import Provider from './provider';
-import Suggestion from './suggestion.jsx';
-
-export type Results = {
-    matchedPretext: string;
-    terms: string[];
-    items: Channel[];
-    component: React.ElementType;
-}
-
-type ResultsCallback = (results: Results) => void;
+import Provider, {ResultsCallback} from './provider';
+import {SuggestionContainer, SuggestionProps} from './suggestion';
 
 type ChannelSearchFunc = (term: string, success: (channels: Channel[]) => void, error?: (err: ServerError) => void) => (ActionResult | Promise<ActionResult | ActionResult[]>);
 
-class ChannelSuggestion extends Suggestion {
-    render() {
-        const isSelection = this.props.isSelection;
-        const item = this.props.item;
-
-        const channelName = item.display_name;
-        const purpose = item.purpose;
-
-        const icon = (
-            <span className='suggestion-list__icon suggestion-list__icon--large'>
-                <i className='icon icon--standard icon--no-spacing icon-globe'/>
-            </span>
-        );
-        let className = 'suggestion-list__item';
-        if (isSelection) {
-            className += ' suggestion--selected';
-        }
-
-        const description = '(~' + item.name + ')';
-
-        return (
-            <div
-                className={className}
-                onClick={this.handleClick}
-                onMouseMove={this.handleMouseMove}
-                {...Suggestion.baseProps}
-            >
-                {icon}
-                <div className='suggestion-list__ellipsis'>
-                    <span className='suggestion-list__main'>
-                        {channelName}
-                    </span>
-                    {description}
-                    {purpose}
-                </div>
+const GenericChannelSuggestion = React.forwardRef<HTMLDivElement, SuggestionProps<Channel>>((props, ref) => {
+    const item = props.item;
```

Comment:
Again coherence between `const {item} = props` and `const item = props.item`.

In general, in mobile we use more the approach of:
```
const MyComponent = ({
    prop1,
    prop2,
}) => {
    return <div />
}
```

0/5 if this is a pattern you want to start following in web.

Reply:
In this case, we need to destructure `props` to pass into the `SuggestionContainer`, so it's unfortunately not an option here